shark_turbine/kernel/compiler/kernel_codegen.py

```diff
@@ -189,6 +190,49 @@ def add_grid(self, grid: Type[Grid]):
                 )
             )
 
+    def determine_input_output_buffers(self, graph: fx.Graph):
+        placeholder_nodes: list[fx.Node] = []
+        for node in graph.nodes:
+            if node.op != "placeholder":
+                continue
+            placeholder_nodes.append(node)
+
+        def only_read_dependencies(node):
+            return all(["read" in x.name for x in node.users.keys()])
+
+        def only_write_dependencies(node):
+            if len(node.users) == 0:
+                return False
+            return all(["write" in x.name for x in node.users.keys()])
+
+        for node in placeholder_nodes:
+            index = None
+            for i, binding in enumerate(self.bindings):
+                if binding.reference[1] == node:
+                    index = i
+                    break
+            if index == None:
+                continue
+            # TODO: remove this hack, this is just to make things pass
+            # I did not investigate yet why it does not correctly determine the
+            # buffer to only have read dependencies, even though that is the case
+            usage = KernelBufferUsage.INPUT
```

Comment:
Original hack from Martin is here https://github.com/harsh-nod/SHARK-Turbine/commit/767f7ecbabace15aa253437b786a89a256fe4d3f#diff-ed8153028c8d651d74c3da869ae799a720faa509ea86828e2beaec1a4fd42ac6R240-R244 

Seems like eventually he added a guard but still default to usage = KernelBufferUsage.INPUT. But I think the right thing to do is set bufferUsage to `KernelBufferUsage.NONE` if it is transient, and if it has both inputs and outputs then it may be transient/temporary buffer. 

Let me double check with Stella on this though.

Reply:
thanks!